lib/blacklight/xml_parser.rb

```diff
@@ -46,19 +39,19 @@ def self.iterate_xml(resources, zip_file)
     resources_array = []
     resources[0].children.each do |resource|
       file_name = resource.attributes["file"].value
-      data_file = Blacklight.open_file(zip_file, file_name)
-      data = Nokogiri::XML.parse(data_file)
-      xml_data = data.children.first
-      type = xml_data.name.downcase
-      if RESOURCE_TYPE[type.to_sym]
-        resource_type = "Blacklight::" + RESOURCE_TYPE[type.to_sym]
-        res_class = resource_type.split("::").
-          reduce(Object) { |o, c| o.const_get c }
-        resource = res_class.new
-        resources_array.push(resource.iterate_xml(xml_data))
+      unless zip_file.find_entry(file_name) == nil
```

Comment:
I try to not use `unless meh == nil`
consider `if zip_file.find_entry(file_name)` instead